src/android/shakeDetector.java

```diff
@@ -0,0 +1,253 @@
+                    //NotificationHelper.createNotification(this, Constants.TRACKING_ERROR_ID, LAccTag2);
+                    Log.i(this, LAccTag2, "," + Float.toString(xAcceleration) + "," + Float.toString(yAcceleration) + "," + Float.toString(zAcceleration) + "," + Float.toString(acceleration2) + ",");
+                    notifyEvent(this, POTENTIAL_INCIDENT, new JSONObject());
+                }
+
+            }
+            else if(event.sensor.getType() == Sensor.TYPE_GYROSCOPE){
+                float xRotation = event.values[0];
+                float yRotation = event.values[1];
+                float zRotation = event.values[2];
+                String GyroTag = "2Gyro5";
+
+                //Log.i(this, GyroTag, ","+String.valueOf(currTime)+"," + Float.toString(xRotation) + "," );
+
+                if ((xRotation > 5f) && (yRotation < 5f) && (zRotation) < 5f) {
+                    lastUpdated = currTime;
+                    //NotificationHelper.createNotification(this, Constants.TRACKING_ERROR_ID, GyroTag);
+                    Log.i(this, GyroTag, "," + Float.toString(xRotation) + "," );
+                    notifyEvent(this, POTENTIAL_INCIDENT, new JSONObject());
+                }
+            }
+        }
+    }
+
+    public void notifyEvent(Context context, String eventName, JSONObject autogenData) {
+        String TAG = "ShakeDetector";
+
+        Log.d(context, TAG, "Generating all notifications for generic "+eventName);
+        try {
+            JSONObject notifyConfigWrapper = UserCacheFactory.getUserCache(context).getLocalStorage(eventName, false);
+            if (notifyConfigWrapper == null) {
+                Log.d(context, TAG, "no configuration found for event "+eventName+", skipping notification");
+                return;
+            }
+            JSONArray notifyConfigs = notifyConfigWrapper.getJSONArray(CONFIG_LIST_KEY);
+            JSONArray mutedConfigs = notifyConfigWrapper.optJSONArray(MUTED_LIST_KEY);
+            for(int i = 0; i < notifyConfigs.length(); i++) {
+                try {
+                    JSONObject currNotifyConfig = notifyConfigs.getJSONObject(i);
+                    int mutedIndex = findEntryWithId(mutedConfigs, currNotifyConfig.getLong(ID));
+                    if(mutedIndex == -1) {
+                        if (autogenData != null) { // we need to merge in the autogenerated data with any user data
+                            JSONObject currData = currNotifyConfig.optJSONObject("data");
+                            if (currData == null) {
+                                currData = new JSONObject();
+                                currNotifyConfig.put("data", currData);
+                            }
+                            mergeObjects(currData, autogenData);
+                        }
+                        Log.d(context, TAG, "generating notification for event "+eventName
+                                + " and id = " + currNotifyConfig.getLong(ID));
```

Comment:
My suggestion is to clone the `currNotifyConfig` here and make the id of the cloned config = orig id + current time or something unique, and then pass the cloned config to be scheduled.

Reply:
This suggestion has not yet been implemented.